### PR TITLE
feat(ml): propagate pipeline config through NATS pull-mode tasks

### LINKED ISSUE #1275: Propagate PipelineRequestConfigParameters through pull-mode (NATS) tasks
Surfaced during PR #1272 occurrence-tracking E2E testing against [ADC PR #77](https://github.com/RolnickLab/ami-data-companion/pull/77).

## Problem

`PipelineRequestConfigParameters` (the `config` dict in `PipelineRequest`) only reaches **sync HTTP `/process`** workers. The NATS pull-mode task schema does not carry it.

`ami/ml/schemas.py::PipelineProcessingTask`:
```python
class PipelineProcessingTask(pydantic.BaseModel):
    id: str
    image_id: str
    image_url: str
    reply_subject: str | None = None
    # TODO: Do we need these?
    # detections: list[DetectionRequest] | None = None
    # config: PipelineRequestConfigParameters | dict | None = None
```

`config` is commented out. `ami/ml/orchestration/jobs.py::queue_all_images_for_job` publishes only the per-image fields. Pull-mode workers cannot see `include_features`, `include_logits`, or any future per-project / per-pipeline flag.

## Why now

PR #1272 needs `include_features=true` to populate `Classification.features_2048` for tracking.
- Sync HTTP: works via `Pipeline.default_config = {"include_features": True}`.
- Pull-mode (ADC): worker must start with `AMI_INCLUDE_FEATURES=true` in its own env. That makes features a worker-level on/off, not per-project, and silently ignores antenna's `ProjectPipelineConfig`.

## Scope

- Add `config: PipelineRequestConfigParameters | dict | None = None` to `PipelineProcessingTask`.
- Populate `task.config` from `pipeline.get_config(project_id=...)` in `ami/ml/orchestration/jobs.py`.
- Update ADC worker (and any other pull-mode worker) to read `config.include_features` from the incoming task; keep env var as fallback.
- Document new contract in `processing_services/README.md` + schema docstring.
- Test: `ProjectPipelineConfig.config={'include_features': true}` reaches the ADC worker via NATS task.

## Risk / migration

Additive + optional. Existing pull-mode workers that ignore `config` keep working; defaults stay safe.

## Linked

- PR #1272 (surfaced this gap)
- RolnickLab/ami-data-companion#77 (uses local env var today)
- Planning doc: `docs/claude/planning/pipeline-config-in-nats-task.md`

### PR DESCRIPTION
**Pull request by mihow**
## Summary

When Antenna runs an async ML job, workers pull tasks from NATS instead of receiving an HTTP request. Until now, those NATS tasks did not carry the pipeline's configuration — only the sync HTTP path did. Any per-project overrides set via `ProjectPipelineConfig` were silently ignored by pull-mode workers, which fell back to their own env vars.

This PR adds a `config` field to the NATS task and result schemas and populates it from `pipeline.get_config(project_id=...)`. It is **structural only** — no new config parameters are introduced, and the value is not yet read during inference. Using config to drive worker behavior is a follow-up.

Also fixes a pre-existing bug in `Pipeline.get_config()` that mutated `self.default_config` in place when a project override was applied, so the in-memory model attribute (and subsequent `get_config()` calls) leaked the override.

Finally, adds a small audit log: when a worker echoes back the config it actually used, Antenna now compares it against the current pipeline config and logs a warning on drift (e.g. `ProjectPipelineConfig` was edited mid-job). Persisted audit trails are a future addition.

- `PipelineProcessingTask` gains `config: PipelineRequestConfigParameters | dict | None` (was commented-out TODO)
- `PipelineResultsResponse` gains matching `config` field so workers can echo back what config was actually used
- `queue_images_to_nats` calls `pipeline.get_config(project_id=job.project.pk)` once per job and embeds the result in each task, so per-project `ProjectPipelineConfig` overrides reach pull-mode workers
- `Pipeline.get_config()` now returns a fresh `PipelineRequestConfigParameters` rather than mutating `self.default_config`
- `process_nats_pipeline_result` warns when echoed config drifts from the pipeline's current configuration
- Documents the NATS task contract in `processing_services/README.md`

## Test plan

- New unit test: `TestJobView.test_queue_images_to_nats_embeds_pipeline_config` — verifies that a `ProjectPipelineConfig` override propagates into the `config` field of each queued `PipelineProcessingTask`
- New unit test: `TestPipeline.test_get_config_does_not_mutate_default_config` — resolves project-scoped config first and asserts `default_config` is untouched
- New unit tests: `test_pipeline_config_drift_logs_warning` and `test_pipeline_config_match_does_not_warn` — verifies the audit log fires on mismatch and stays silent on match
- Existing tasks endpoint tests unchanged

## Follow-ups

A short proposal for moving the per-task `config` field to a job-level fetch (since all tasks in a job share the same config) lives at [`docs/claude/planning/pipeline-config-job-level-fetch.md`](https://github.com/RolnickLab/antenna/blob/worktree-job-results-response-cleanup/docs/claude/planning/pipeline-config-job-level-fetch.md). Worth doing before configs grow to include taxa allow-lists or per-stage hyperparameters.

## Companion PR

RolnickLab/ami-data-companion#146 — ADC schema update to receive and echo config

Closes #1275

## Summary by CodeRabbit

* **New Features**
  * Per-task pipeline configuration is now included with queued async jobs and echoed with processing results.

* **Bug Fixes**
  * Pipeline default config is no longer mutated when resolving project overrides.

* **Tests**
  * Added unit and E2E-style tests verifying config propagation, echoing, and drift detection behavior.

* **Documentation**
  * Expanded async processing docs describing task-level config, expected worker behavior, and drift auditing.

### CODE REVIEW

**Comment by mihow** [comment]
**Claude says:** (posted on Michael's behalf — part of a structural review pass over the four open PSv2-readiness PRs: #1276, #1279, #1312, #1324, toward the umbrella #910.)

The core design here looks right: snapshotting `pipeline.get_config(project_id=...)` once at enqueue and embedding it in every task message makes the contract self-contained — no out-of-band fetch, no race with config edits mid-job, and old in-flight messages parse fine since the field is optional. The `get_config()` mutation fix is a quietly excellent catch (the old `self.default_config` + `update()` leaked project overrides into the in-memory model for the sync path too), and it ships with a regression test. We also verified `queue_images_to_nats` is the single NATS publish path, so coverage is complete.

Three things seem worth resolving before merge:

1. **The drift audit will likely warn on every result batch from currently-deployed workers.** The ADC `PipelineResultsResponse` on its main branch already defaults `config` to a `PipelineConfigRequest()` and serializes without `exclude_none`, so every result echoes `{"example_config_param": null}`. Antenna compares that against `dict(get_config(...))`, which is `{}` for pipelines with no default config — so `{} != {"example_config_param": None}` fires `job.logger.warning` once per batch on every async job (now persisted to the JobLog table from #1259), plus an extra config query per result. The drift tests only exercise symmetric hand-built dicts, so this shape isn't covered. Suggestion: normalize both sides (drop `None` values) before comparing, or skip the audit when the echoed config is empty/all-None.

2. **Worth keeping #1275 open after this merges.** The companion ADC PR receives and echoes the config, but its `PipelineConfigRequest` schema has a single example field and pydantic silently drops unknown keys — so `include_features`, the parameter #1275 was filed for, never actually arrives at the inference code. Both PRs say "structural only" honestly; just flagging that a third change (real fields in the ADC schema + applying them in inference) is needed before the feature exists end-to-end, and the silent-drop behavior means there's currently no signal when a key doesn't make it across.

3. **Contract asymmetry as a future foot-gun:** Antenna's side is an open dict that accepts anything; ADC's side is a strict model that silently discards unknowns; there's no version field or rejected-key feedback. The echo-audit was presumably meant to be the safety net for exactly this, which makes item 1 worth fixing rather than muting.

Smaller notes: the branch has been open since April and main has moved (some touched functions have shifted), so a rebase + test re-run is worth doing; and a couple of the bot threads (drift test running through a failing save path, README wording about "recording" vs logging the echoed config) still look open.

The architecture is the right one — with the audit noise fixed and #1275 kept open as the tracking issue for the apply-side, this seems good to go.
